File: 1. Practice/(a) Puzzles/i. Classic Puzzle - Easy/Power of Thor - Episode 1/Power of Thor - Episode 1.cs
// Seonghyeon Choe
// schoe@seonghyeonchoe.com
// 2021

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;

class Player
{
    static void Main(string[] args)
    {
        string[] inputs = Console.ReadLine().Split(' ');
        int lightX = int.Parse(inputs[0]); // the X position of the light of power
        int lightY = int.Parse(inputs[1]); // the Y position of the light of power
        int initialTx = int.Parse(inputs[2]); // Thor's starting X position
        int initialTy = int.Parse(inputs[3]); // Thor's starting Y position

        // game loop
        while (true)
        {
            int remainingTurns = int.Parse(Console.ReadLine()); // The remaining amount of turns Thor can move. Do not remove this line.

            String direction = "";
            if (initialTy > lightY )
            {
                direction = direction + "N";
                --initialTy;
            }
            else if (initialTy < lightY )
            {
                direction = direction + "S";
                ++initialTy;
            }

            if (initialTx > lightX )
            {
                direction = direction + "W";
                --initialTx;
            }
            else if (initialTx < lightX )
            {
                direction = direction + "E";
                ++initialTx;
            }

            Console.WriteLine(direction);
        }
    }
}

File: 1. Practice/(a) Puzzles/i. Classic Puzzle - Easy/Power of Thor - Episode 1/Power of Thor - Episode 1.cpp
// Seonghyeon Choe
// schoe@seonghyeonchoe.com
// 2021

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;
 
int main()
{
    int lightX; // the X position of the light of power
    int lightY; // the Y position of the light of power
    int initialTX; // Thor's starting X position
    int initialTY; // Thor's starting Y position
    cin >> lightX >> lightY >> initialTX >> initialTY; cin.ignore();

    // game loop
    while (1) {
        int remainingTurns; // The remaining amount of turns Thor can move. Do not remove this line.
        cin >> remainingTurns; cin.ignore();

        if (initialTY > lightY )
        {
            cout << "N";
            --initialTY;
        }
        else if (initialTY < lightY )
        {
            cout << "S";
            ++initialTY;
        }

        if (initialTX > lightX )
        {
            cout << "W";
            --initialTX;
        }
        else if (initialTX < lightX )
        {
            cout << "E";
            ++initialTX;
        }

        cout << endl;
    }
}